src/gui/scene/scene.cpp

```diff
@@ -0,0 +1,112 @@
+#include "scene.hpp"
+
+#include "../../entity/mesh.hpp"
+#include "../data_structures/static_mesh.hpp"
+#include "../data_structures/terrain_mesh.hpp"
+#include "../handler.hpp"
+
+#include <GLFW/glfw3.h> // Will drag system OpenGL headers
+
+gui::Scene::Scene(
+    World& world, uint32_t window_width, uint32_t window_height,
+    uint32_t shadow_map_width_height
```

Comment:
Typedefs

Reply:
I don't want to change many things in gui because it will be more work when merging with Graphics/refactoring